Sources/M3UKit/PlaylistParser.swift

```diff
@@ -77,7 +77,7 @@ public final class PlaylistParser {
 
       if self.isInfoLine(line) {
         lastMetadataLine = line
-      } else if let url = URL(string: line) {
+      } else if let url = URL(string: line.trimmingCharacters(in: .whitespaces)) {
```

Comment:
Wondering what is the cost for repeating this operation at for all items in large playlists 🤔

Reply:
Have been using for a while in my app and it doesn't look it impacts the performance, but feel free to run performance tests if you want to be sure.

Reply:
With a file containing end of line characters, I am getting a fail too and your fix doesn't help strangely, even when I use .controlCharacters

691775^M

Would you have any ideas ?

thanks

Reply:
Could you provide an example?

Reply:
yeah sure, I have actually tracked down to the playlist having a first empty line (and the control characters which I can handle with your fix)

example, which causes an error.

https://yoyo.org/~jt/random/test.m3u

Reply:
Im gonna check if I can find a nice solution for it

Reply:
Hey, sorry, I completely forgot about you, but indeed the way you are handling the issue looks ideal, cleaning the string before trying to parse, I don't have a better solution for now.
It annoys me a lot that IPTV providers have these discrepancies

Reply:
Hey, hey 

no worries. I'ma actually cleaning the whole file first. I could try a hack where it cleans it line by line I suppose (within M3UKit)